src/tir/schedule/primitive/fuse_split.cc

```diff
@@ -0,0 +1,483 @@
+    } else {
+      if (infer_index != -1) {
+        throw NotSingleInferFactorError(self->mod);
+      } else {
+        infer_index = i;
+      }
+    }
+  }
+  // Step 2. infer factors if needed
+  Array<PrimExpr> inferred_factors(factors);
+  if (infer_index != -1) {
+    inferred_factors.Set(infer_index,
+                         analyzer.Simplify(floordiv(loop->extent + tot_length - 1, tot_length)));
+  } else {
+    if (!analyzer.CanProve(tot_length >= loop->extent)) {
+      throw WrongFactorProductError(self->mod, GetRef<For>(loop));
+    }
+  }
+  // Step 3. Replace all occurrence of the original loop var with new variables
+  std::vector<Var> new_loop_vars;
+  new_loop_vars.reserve(inferred_factors.size());
+  for (size_t i = 0; i < inferred_factors.size(); i++) {
+    new_loop_vars.push_back(loop->loop_var.copy_with_suffix("_" + std::to_string(i)));
+  }
+  PrimExpr substitute_value = 0;
+  for (size_t i = 0; i < inferred_factors.size(); i++) {
+    substitute_value *= inferred_factors[i];
+    substitute_value += new_loop_vars[i];
+  }
+  Map<Block, Block> opaque_block_reuse;
+  auto substitute_function = [&](const Var& v) -> Optional<PrimExpr> {
+    if (v.same_as(loop->loop_var)) {
+      return substitute_value;
+    } else {
+      return NullOpt;
+    }
+  };
+  Stmt new_loop_body =
+      SubstituteAndCollectOpaqueBlock(loop->body, &opaque_block_reuse, substitute_function);
+  for (size_t i = 0; i < inferred_factors.size(); i++) {
+    analyzer.Bind(new_loop_vars[i], Range::FromMinExtent(0, inferred_factors[i]));
+  }
+  // Step 4. Update predicate to guard the loop
+  PrimExpr predicate = substitute_value < loop->extent;
+  new_loop_body = PredicateUpdater(predicate, &analyzer)(new_loop_body);
+  // Step 5. Generate tnested loops to replace the original loop and simplify the binding
+  Stmt outer_stmt = new_loop_body;
+  for (int i = inferred_factors.size() - 1; i >= 0; i--) {
+    outer_stmt = For(new_loop_vars[i], 0, inferred_factors[i], loop->kind, outer_stmt);
+  }
+
+  outer_stmt =
+      Downcast<For>(SimplifyBindings(outer_stmt, GetLoops(loop_sref), &opaque_block_reuse));
+  self->Replace(loop_sref, outer_stmt, opaque_block_reuse);
+  Array<StmtSRef> result_srefs;
+  result_srefs.reserve(inferred_factors.size());
+  for (size_t i = 0; i < inferred_factors.size(); i++) {
+    result_srefs.push_back(self->stmt2ref.at(outer_stmt.get()));
+    const ForNode* outer_loop = outer_stmt.as<ForNode>();
+    ICHECK(outer_loop);
+    outer_stmt = outer_loop->body;
+  }
+  return result_srefs;
+}
+
+StmtSRef Fuse(ScheduleState self, const Array<StmtSRef>& loop_srefs) {
+  //     Invariance
+  //   - The total repeat number has not changed for each direct child block.
+  //   - The execution order has not changed. (The block executes with the same
+  //     args and the same order with before.)
+  std::vector<const ForNode*> loops;
+  loops.reserve(loop_srefs.size());
+  StmtSRef outer_sref{nullptr};
```

Comment:
let's be consistent with the naming:
- block_sref: An sref to a block
- loop_sref: An sref to for a loop
- block_srefs: An array of block srefs
- loop_srefs: An array of loop srefs
- sref: An sref to either a block or a loop (either way is possible)

```suggestion
  StmtSRef outer_loop_sref{nullptr};
```